src/responder/kcm/kcmsrv_ccache_secdb.c

```diff
@@ -59,7 +59,20 @@ static errno_t sec_get(TALLOC_CTX *mem_ctx,
         goto done;
     }
 
-    buf = sss_iobuf_init_steal(tmp_ctx, data, len);
+    if (strcmp(datatype, "simple") != 0) {
+        buf = sss_iobuf_init_steal(tmp_ctx, data, len);
+    } else {
+        /* The secret is stored in b64 encoding, we need to decode it first. */
+        data = sss_base64_decode(tmp_ctx, (const char*)data, &len);
+        if (data == NULL) {
+            DEBUG(SSSDBG_CRIT_FAILURE, "Cannot decode secret from base64\n");
+            ret = EIO;
+            goto done;
+        }
+
+        buf = sss_iobuf_init_readonly(tmp_ctx, data, len);
```

Comment:
This could be `sss_iobuf_init_steal()` as well but I'm ready to ACK both versions.

Reply:
Done. Thank you.